main.go

```diff
@@ -279,19 +290,26 @@ func doMain(args []string) error {
 		}
 
 	case *runOptions.daemon:
-		controler := NewMender(env)
-		if err := controler.LoadConfig(*runOptions.config); err != nil {
-			return err
+		updater, err := NewUpdateClient(config.GetUpdaterConfig())
+		if err != nil {
+			return errors.New("Cannot initialize daemon. Error instantiating updater. Exiting.")
+		}
+
+		controler := NewMender(*config, MenderPieces{
```

Comment:
Not sure if this makes sense now, but maybe split into two: `MenderServerIface` and `MenderHWIface`? Then HW will be everything related to interactions with device and server will be communication part?